### PR TITLE
chore(docs): Fix some typos

### PR DESCRIPTION
**Pull request by nikeee**
# Summary
Fixes some typos in react compiler docs

# How did you test this change?
No tests needed

### CODE REVIEW

**Comment by nikeee** [comment]
Rebased to current main

File: compiler/docs/DESIGN_GOALS.md [hunk 1]
```diff
@@ -42,11 +42,13 @@ React Compiler has two primary public interfaces: a Babel plugin for transformin
...
-- **Type Inference** (InferTypes): We run a conservative type inference pass to identify certain key types of data that may appear in the program that are relevant for further analysis, such as which values are hooks, primitives, etc. 
+- **Type Inference** (InferTypes): We run a conservative type inference pass to identify certain key types of data that may appear in the program that are relevant for further analysis, such as which values are hooks, primitives, etc.
+- **Inferring Reactive Scopes**: Several passes are involved in determining groups of values that are created/mutated together and the set of instructions involved in creating/mutating those values. We call these groups "reactive scopes", and each can have one or more declarations (or occasionally a reassignment).
+- **Type Inference** (InferTypes): We run a conservative type inference pass to identify certain key types of data that may appear in the program that are relevant for further analysis, such as which values are hooks, primitives, etc.
```
Comment (josephsavona) [inline compiler/docs/DESIGN_GOALS.md:51]: looks like this got duplicated

**Review by josephsavona — COMMENTED** [review]
@nikeee sorry that we missed this earlier, the changes look good except for one place where text got duplicated (bad rebase maybe?). thank you for sending in these fixes!

**Comment by nikeee** [comment]
Yep, a bad rebase, oopsie. Should be fixed

**Pull request closed without merging**